salt/metalk8s/orchestrate/upgrade/etcd.sls

```diff
@@ -0,0 +1,52 @@
+{%- set dest_version = pillar.orchestrate.dest_version %}
+{%- set etcd_nodes = salt.metalk8s.minions_by_role('etcd') %}
+{%- set control_plane_ip = salt.saltutil.runner(
+    'mine.get',
+    tgt=pillar.bootstrap_id,
+    fun='control_plane_ip'
+)[pillar.bootstrap_id] %}
+
+{%- set pillar_data = {
+    'repo': {
+        'host': control_plane_ip
+    },
+    'registry_ip': control_plane_ip
+} %}
+
+Check etcd cluster health:
+  module.run:
+    - metalk8s.check_etcd_health:
+      - hostname: {{ pillar.bootstrap_id }}
+    - failhard: True
+
+{%- for node in etcd_nodes %}
```

Comment:
We should sort the nodes somehow.

Rationale: if one runs this playbook, and it fails, and for whatever reason the playbook is executed again without the cause for failure being fixed, then we want to fail at the same point, rather than potentially first upgrading other nodes that weren't upgraded before, to then at some point bump into the failing node.

Reply:
Hmm yes ok make sense  i can sort using node name